frontend/src/components/FundsPullPreApproval/ApprovalDetails.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from "react";
+import { Approval } from "../../interfaces/approval";
+import { diemAmountToHumanFriendly } from "../../utils/amount-precision";
+import { settingsContext } from "../../contexts/app";
+
+interface RequestDetailsProps {
+  approval: Approval | undefined;
+}
+
+function ApprovalDetails({ approval }: RequestDetailsProps) {
+  const [settings] = useContext(settingsContext)!;
+
+  const dateToDisplayByApprovalStatus = () => {
+    if (approval!.status === "pending") {
+      return (
+        <>
+          {"on"} <span>{new Date(approval!.created_at).toLocaleString()}</span>
+        </>
+      );
+    }
+    if (approval!.status === "rejected") {
+      return (
+        <div>
+          {"Rejected on"} <span>{new Date(approval!.updated_at).toLocaleString()}</span>
+        </div>
+      );
+    }
+    if (approval!.status === "closed") {
+      return (
+        <>
+          <div>
+            {"Approved on"} <span>{new Date(approval!.approved_at).toLocaleString()}</span>
+          </div>
+          <div>
+            {"Revoked on"} <span>{new Date(approval!.updated_at).toLocaleString()}</span>
+          </div>
+        </>
+      );
+    }
+    if (approval!.status === "valid") {
+      const now = new Date().toISOString();
+      if (Date.parse(approval!.scope.expiration_timestamp) < Date.parse(now)) {
+        return (
+          <>
+            <div>
+              {"Approved on"} <span>{new Date(approval!.approved_at).toLocaleString()}</span>
+            </div>
+            <div>
+              {"Expired on"}{" "}
+              <span>{new Date(approval!.scope.expiration_timestamp).toLocaleString()}</span>
+            </div>
+          </>
+        );
+      }
+
+      return (
+        <div>
+          {"Approved on"} <span>{new Date(approval!.updated_at).toLocaleString()}</span>
+        </div>
+      );
+    }
+  };
+
+  return (
+    <span>
+      <div className="text-black">
+        <span>
+          Received from <b>{approval!.biller_name}</b>
```

Comment:
You defined `approval` to be optional, but treat it like it's always there. Either treat the case for when it wasn't provided, or make it required and remove the `!`